src/Infrastructure/LeanCode.Logging/LeanCodeLoggingBuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using LeanCode.Logging;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class LeanCodeLoggingBuilderExtensions
+{
+    public static ILoggingBuilder ClearLeanCodeLoggers(this ILoggingBuilder builder)
+    {
+        builder.Services.RemoveAll(typeof(LeanCode.Logging.ILogger<>));
+
+        return builder;
+    }
+
+    public static ILoggingBuilder TryAddContextualLeanCodeLogger(this ILoggingBuilder builder)
+    {
+        builder.Services.TryAddSingleton(typeof(LeanCode.Logging.ILogger<>), typeof(ContextualLogger<>));
```

Comment:
Transient, not singleton - otherwise the loggers will stay instantiated even if the class using them will be long gone (think rarely invoked handlers).

Reply:
Wouldn't that cause DI validation to complain if we try to resolve this logger in a service registered as singleton?

And even if not, wouldn't Scoped be better than Transient? I don't see a reason to instanciate the logger for the same context multiple times per request.

Reply:
Good point, dunno how to tackle the singleton problem. :( Nevertheless, the main usecase is "quickly instantiate logger for a very short time", so would go with it either way (which also would work the same way as previously when it comes to GC/instantiation).

Ad. Scoped/Transient - yep, you're right. I forgot the nomenclature.